### PR TITLE
feat: adding warehouse filter for gross profit report

### PR DESCRIPTION
**Pull request by Gokulnath17**
In gross profit report, there is no warehouse filter, so i will add warehouse filter in gross profit report report.

**Uses:** If some company use warehouse for their branch, so warehouse filter is used for that types companies and this feature also used for all types of companies.

image

### CODE REVIEW

File: erpnext/accounts/report/gross_profit/gross_profit.js [hunk 1]
```diff
@@ -50,6 +50,12 @@ frappe.query_reports["Gross Profit"] = {
 			"fieldtype": "Link",
 			"options": "Sales Person"
 		},
+		{
+			"fieldname": "warehouse",
+			"label": __("Warehouse"),
+			"fieldtype": "Link",
+			"options": "Warehouse"
+		},
```
Comment (ruthra-kumar) [inline erpnext/accounts/report/gross_profit/gross_profit.js:66]: 1. Use `get_query` to filter Warehouses on Company. Currently it shows warehouses from all the companies on a multi company setup. `get_query` ex: https://github.com/frappe/erpnext/blob/a7b011aed4c879e077a3a3e81c6834dc2555eac1/erpnext/accounts/report/accounts_payable_summary/accounts_payable_summary.js#L65-L73
3. When a root node is selected, all of its child nodes should be included in the filter. You can use the utlity function for that. https://github.com/frappe/erpnext/blob/a7b011aed4c879e077a3a3e81c6834dc2555eac1/erpnext/stock/report/stock_ledger/stock_ledger.py#L430

**Comment by deepeshgarg007** [comment]
@Gokulnath17 Do you plan to fix this? Else we will close the PR

File: erpnext/accounts/report/gross_profit/gross_profit.js (hunk 1 above)
Comment (Gokulnath17) [reply erpnext/accounts/report/gross_profit/gross_profit.js:66]: Now fixed kindly check it...

> @Gokulnath17 Do you plan to fix this? Else we will close the PR
Updated

Comment (ruthra-kumar) [reply erpnext/accounts/report/gross_profit/gross_profit.js:66]: > When a root node is selected, all of its child nodes should be included in the filter. You can use the utlity function for that.

@Gokulnath17  This is not fixed

**Pull request merged**